extensions/cornerstone/src/services/SegmentationService/SegmentationService.ts

```diff
@@ -539,13 +539,29 @@ class SegmentationService extends PubSubService {
       throw new Error(`No volume found for referencedVolumeId: ${referencedVolumeId}`);
     }
 
+    const useSharedArrayBufferValue = window.config?.useSharedArrayBuffer;
+    let sharedArrayBufferEnabled = true;
+
+    if (useSharedArrayBufferValue === 'AUTO') {
+      try {
+        /*eslint-disable no-constant-condition */
+        if (!new SharedArrayBuffer(0)) {
```

Comment:
All of this code to detect the flag should be isolated to a separate function, which stores the result of the test, so that on subsequent calls it returns immediately.  That test should be in the platform/core functionality as it is shared functionality.